src/Mailgun/Api/HttpApi.php

```diff
@@ -56,38 +56,58 @@ public function __construct(HttpClient $httpClient, RequestBuilder $requestBuild
     }
 
     /**
-     * Attempts to safely deserialize the response into the given class.
-     * If the HTTP return code != 200, deserializes into SimpleResponse::class
-     * to contain the error message and any other information provided.
-     *
      * @param ResponseInterface $response
-     * @param string            $className
+     * @param string            $class
      *
-     * @throws HttpClientException
+     * @return mixed|array|ResponseInterface
```

Comment:
Technically. We do not know at all what to return. Since we do not know what the Hydrator return. Maybe it should be `mixed|ResponseInterface` since that is the only thing we are sure at.